src/components/reviews/addReviewForm.jsx

```diff
@@ -57,6 +59,55 @@ class AddReviewForm extends React.Component {
     });
   }
 
+  readFileAsync(file) {
+    return new Promise((resolve, reject) => {
+      const reader = new FileReader();
+      reader.onload = () => { resolve(reader.result) };
+      reader.readAsDataURL(file);
+    });
+  }
+
+  uploadChange(event) {
+
+    if (!event.target.files || !event.target.files[0]) { return; }
+
+    const file = event.target.files[0];
+
+    if (!file.type.startsWith("image/")) {
+      this.setState({
+        imageError: "You may only upload images."
+      });
+    } else if (this.state.photos.length >= 5) {
+      this.setState({
+        imageError: "Photo limit reached."
+      });
+    } else {
+
+      this.setState({
+        imageError: ""
+      });
+
+      this.readFileAsync(file)
+        .then((data) => {
+          axios.post("https://api.cloudinary.com/v1_1/greenfield/image/upload", {
```

Comment:
Really impressed how you found this image hosting platform and did all of this work. Shows a quick ability to synthesize new topics.